pkgs/by-name/od/odmpy/package.nix

```
@@ -0,0 +1,86 @@
+{
+  ffmpeg,
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  writableTmpDirAsHomeHook,
+}:
+
+let
+  version = "0.8.1";
+in
+python3Packages.buildPythonApplication {
+  pname = "odmpy";
+  inherit version;
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ping";
+    repo = "odmpy";
+    tag = version;
+    hash = "sha256-RWaB/W8ilAKRr0ZSISisCG8Mdgw5LXRCLOl5o1RsmbA=";
+  };
+
+  build-system = with python3Packages; [
+    setuptools
+    wheel
+  ];
+
+  postPatch = ''
+    substituteInPlace odmpy/processing/shared.py \
+      --replace-fail '"ffmpeg",' '"${lib.getExe ffmpeg}",'
+
+    substituteInPlace tests/odmpy_dl_tests.py \
+      --replace-fail '"ffprobe",' '"${lib.getExe' ffmpeg "ffprobe"}",'
+  '';
+
+  dependencies = with python3Packages; [
+    beautifulsoup4
+    eyed3
+    iso639-lang
+    lxml
+    mutagen
+    requests
+    termcolor
+    tqdm
+    typing-extensions
+  ];
+
+  nativeCheckInputs =
+    with python3Packages;
+    [
+      coverage
+      responses
+      ebooklib
+    ]
+    ++ [
+      writableTmpDirAsHomeHook
+    ];
```

```suggestion
  nativeCheckInputs =
    (with python3Packages;
    [
      coverage
      responses
      ebooklib
    ])
    ++ [
      writableTmpDirAsHomeHook
    ];
```
to properly isolate scopes